docs/esbuild.md

```diff
@@ -22,31 +22,30 @@ or using yarn
 yarn add -D @bazel/esbuild
 ```
 
-The esbuild binary is fetched from npm automatically and exposed via toolchains. Add the `esbuild_repositories` rule to the `WORKSPACE`:
+The esbuild binary is fetched automatically for your platform and is exposed via Bazel toolchains.
+To do this, add the `esbuild_repositories` rule to your `WORKSPACE`.
+You'll need to point it to the repository created by npm_install or yarn_install where the `@bazel/esbuild`
+package is being fetched. (Typically, this is `npm`).
+Set the `npm_repository` attribute to the name of that repository.
 
 ```python
-load("@npm//@bazel/esbuild:esbuild_repositories.bzl", "esbuild_repositories")
-
-esbuild_repositories()
-```
-
-As esbuild is being fetched from `npm`, the load statement above can cause eager fetches of the `@npm` external repository.
-To work around this, it's possible to fetch the `@bazel/esbuild` package via an `http_archive`
-
-```python
-http_archive(
-    name = "bazel_esbuild",
-    urls = [
-        "https://registry.npmjs.org/@bazel/esbuild/-/esbuild-4.0.0.tgz",
-    ],
-    strip_prefix = "package",
+npm_install(
+    name = "npm",
+    # @bazel/esbuild is a dependency in this package.json
+    package_json = "//:package.json",
+    package_lock_json = "//:package-lock.json",
 )
 
-load("@bazel_esbuild//:esbuild_repositories.bzl", "esbuild_repositories")
+load("@npm//@bazel/esbuild:esbuild_repositories.bzl", "esbuild_repositories")
```

Comment:
This should be the rules_nodejs repo load not npm right?

Reply:
oops, where did that come from? I thought I copy-pasted from the working example...

Reply:
oh, I bet I changed the wrong file and when accepting the new golden docs I overwrote it. Almost did it again :)

Reply:
I see, it's getting auto-replaced when we publish the docs, because we always assume the packages are referenced from npm... will have to inhibit that replacement here somehow